From 6ab3b5b7ac9478592655fc437fa7136d649636bc Mon Sep 17 00:00:00 2001
From: Sai Hari Chandana Kalluri <chandana.kalluri@xilinx.com>
Date: Wed, 5 May 2021 09:26:51 -0700
Subject: [PATCH] make-initrd: cross install tweaks

Signed-off-by: Sai Hari Chandana Kalluri <chandana.kalluri@xilinx.com>

---
 initrd/make-initrd | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/initrd/make-initrd b/initrd/make-initrd
index 21170e8..559df30 100755
--- a/initrd/make-initrd
+++ b/initrd/make-initrd
@@ -47,7 +47,22 @@ mkdir -p $tmpdir/sys
 mkdir -p $tmpdir/lib
 mkdir -p $tmpdir/var
 mkdir -p $tmpdir/mnt
-cp "$busybox" $tmpdir/bin/busybox
+
+if [ -z "$BUSYBOX" ]; then
+    BUSYBOX=`which busybox`
+fi
+cp $BUSYBOX $tmpdir/bin
+if [ -n "$CROSS_COMPILE" ]; then
+    echo "cross compiling, busybox --install emulation"
+    if [ -n "$QEMU_USER" ]; then
+        $QEMU_USER $tmpdir/bin/busybox --install $tmpdir/bin
+    else
+        echo "QEMU_USER is not defined, no binary symlinks will be available"
+    fi
+else
+    $tmpdir/bin/busybox --install $tmpdir/bin
+fi
+
 for i in `cat "$base"/applet-list`
 do
     ln -s /bin/busybox $tmpdir/bin/$i
