From 204644c878d33097831ae0f7527f5a933d8059bc Mon Sep 17 00:00:00 2001
From: Manjukumar Matha <manjukumar.harthikote-matha@xilinx.com>
Date: Wed, 21 Mar 2018 22:42:09 -0700
Subject: [PATCH] Upstream-Status: Inappropriate [poky-specific fix]

Signed-off-by: Manjukumar Matha <manjukumar.harthikote-matha@xilinx.com>
---
 agent/Makefile.inc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/agent/Makefile.inc b/agent/Makefile.inc
index f9f191b..15cf390 100644
--- a/agent/Makefile.inc
+++ b/agent/Makefile.inc
@@ -92,6 +92,9 @@ ifeq ($(OPSYS),GNU/Linux)
     OPTS += -DUSE_uuid_generate=0
   endif
   OPTS += -DENABLE_arch_$(shell uname -m)
+  ifneq ($(RANLIB),)
+      RANLIB += $@
+  endif
 endif
 
 ifneq ($(OPSYS),Windows)
-- 
2.7.4

