From b34e8552e6d215c54cc8f3612bf248b8c4887275 Mon Sep 17 00:00:00 2001
From: Sai Hari Chandana Kalluri <chandana.kalluri@xilinx.com>
Date: Wed, 20 Mar 2019 18:59:01 -0700
Subject: [PATCH 2/4] Upgrade rgb lcd example to python3

---
 rgb_lcd_demo/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rgb_lcd_demo/Makefile b/rgb_lcd_demo/Makefile
index 50b60da..429271a 100644
--- a/rgb_lcd_demo/Makefile
+++ b/rgb_lcd_demo/Makefile
@@ -1,4 +1,4 @@
-LDFLAGS=-lupm-i2clcd -lupm-jhd1313m1
+LDFLAGS=-lupm-lcd -lupm-jhd1313m1
 CXXFLAGS=-Wall -g -I/usr/include/upm
 TARGETS=rgb_lcd_demo
 
-- 
2.7.4

